app/javascript/components/applications/ApplicationPage.js

```diff
@@ -7,33 +7,38 @@ import mapProps from '~/utils/mapProps'
 import { mapApplication } from '~/components/mappers/soqlToDomain'
 import labelMapperFields from './application_details/applicationDetailsFieldsDesc'
 
-const buildActionLinkIfNecessary = (app) => {
-  if (!app.isLotterComplete && app.submissionType === 'Paper') {
-    return { title: 'Edit Application', link: appPaths.toApplicationEdit(app.id) }
-  } else {
-    return undefined
-  }
+const buildActionLinkIfNecessary = (app, fromPage) => {
+  const actions = []
+
+  if (!app.is_lottery_complete && app.application_submission_type === 'Paper')
+    actions.push(<a key='edit-application' href={appPaths.toApplicationEdit(app.id)} className='primary button tiny '>Edit Application</a>)
+
+  if (fromPage === 'new')
```

Comment:
are we adding this button if we edit an application too? is "new" a good word to use then?

Reply:
Not just when we come from adding a new application. Not sure about the word, it was the first thing that came to my mind. any thoughts? it could be just a flag so there is no dependency with the previous page. showAddNewButton=true